### ISSUE TITLE
populated virtuals

### ISSUE THREAD

**Issue by PsychicCat** [vkarpov15/mongoose-lean-virtuals#2]
I'm not able to use virtuals on populated models. I'm guessing that is not possible yet?

**Comment by vkarpov15** [comment]
Can you please clarify with some code samples? I don't understand your question

**Comment by facupedrazzini** [comment]
```groups
		.find({})
		.populate('owner')
		.lean({ virtuals: true })
		.exec()```

**Comment by vkarpov15** [comment]
I still don't understand, can you provide schemas?

**Comment by tahasamad** [comment]
```
const mongoose = require("mongoose");
const Schema = mongoose.Schema;
const mongooseLeanVirtuals = require('mongoose-lean-virtuals');

const UserSchema = new Schema({
    firstname: {
        type: String
    },
    lastname: {
        type: String
    },
    bff: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        default: null
    }
}, {
    toObject: {
        virtuals: true
    },
    toJSON: {
        virtuals: true
    }
});

UserSchema.virtual("fullname").get(function() {
    if (this._fullname === undefined) {
        this._fullname = this.firstname + this.lastname;
    }
    return this._fullname;
});

UserSchema.plugin(mongooseLeanVirtuals);

const User = mongoose.model('User', UserSchema);

mongoose.connect("mongodb://localhost/test", {
    useMongoClient: true
}, function(err) {
    if (err) {
        console.log(err);
        return;
    }
    User.findOne({firstname: "Taha"}).populate("bff").lean({virtuals: true}).exec(function(err, user) {
        console.log(user.fullname);
        console.log(user.bff.firstname);
        console.log(user.bff.fullname);//**ISSUE:**Will still be undefined.
    });
});
```

**Comment by vkarpov15** [comment]
Bug with mongoose, will be fixed in mongoose 4.13.8 :+1:

**Comment by arpitsr91** [comment]
I know this is closed , but I'm using mongoose v5 and i still have this problem. I'm using the same syntax as above - Model.find().populate().lean({virtuals:true}).exec(). Im getting all the 'get' virtuals, but neither of the populate virtuals. foreg.  as @tahasamad mentioned, I'm getting user.fullname but not user.bff.fullname. They just come as null. If i remove lean however , i'm getting both.

**Comment by vkarpov15** [comment]
@arpitsr91 just released [0.3.5 with the fix](http://plugins.mongoosejs.io/plugins/lean-virtuals#0-3-5-2019-03-11)
